src/redis/windowed_cache_impl.go

```diff
@@ -0,0 +1,263 @@
+package redis
+
+import (
+	"math"
+	"math/rand"
+
+	"github.com/coocood/freecache"
+	pb "github.com/envoyproxy/go-control-plane/envoy/service/ratelimit/v3"
+	"github.com/envoyproxy/ratelimit/src/assert"
+	"github.com/envoyproxy/ratelimit/src/config"
+	"github.com/envoyproxy/ratelimit/src/limiter"
+	"github.com/envoyproxy/ratelimit/src/utils"
+	"github.com/golang/protobuf/ptypes/duration"
+	logger "github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+)
+
+type windowedRateLimitCacheImpl struct {
```

Comment:
Can you add some high level comments on how rolling window limits are implemented? It will help the reader and my own review.

Reply:
Adding some explanations and reference for GCRA